File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/AnnouncementNoticeTemplateFragmentService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.AnnouncementNoticeTemplateFragment;

public interface AnnouncementNoticeTemplateFragmentService {
    public List<AnnouncementNoticeTemplateFragment> getAnnouncementNoticeTemplateFragments();

    public AnnouncementNoticeTemplateFragment getAnnouncementNoticeTemplateFragment(Integer id);

    public AnnouncementNoticeTemplateFragment addOrUpdateAnnouncementNoticeTemplateFragment(AnnouncementNoticeTemplateFragment announcementNoticeTemplateFragment);
}

File: client/src/app/modules/quotation/services/announcement.notice.template.fragment.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { AnnouncementNoticeTemplateFragment } from '../../quotation/model/AnnouncementNoticeTemplateFragment';

@Injectable({
  providedIn: 'root'
})
export class AnnouncementNoticeTemplateFragmentService extends AppRestService<AnnouncementNoticeTemplateFragment>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getAnnouncementNoticeTemplateFragments() {
    return this.getList(new HttpParams(), "announcement-notice-template-fragment");
  }
  
   addOrUpdateAnnouncementNoticeTemplateFragment(announcementNoticeTemplateFragment: AnnouncementNoticeTemplateFragment) {
    return this.addOrUpdate(new HttpParams(), "announcement-notice-template-fragment", announcementNoticeTemplateFragment, "Enregistré", "Erreur lors de l'enregistrement");
  }

}
